plugins/modules/inventory_intent.py

```diff
@@ -1109,6 +1110,41 @@ def export_device_details(self):
 
         return self
 
+    def check_ap_devices(self, device_ips):
+        """
+        Args:
+            self (object): An instance of a class used for interacting with Cisco DNA Center.
+            device_ip (str): The management IP address of the device for which the response is to be retrieved.
+        Returns:
+            list: A list containing Access Point device IP's obtained from the Cisco DNA Center.
+        Description:
+            This method communicates with Cisco DNA Center to retrieve the details of a device with the specified
+            management IP address and check if device family matched to Unified AP. It executes the 'get_device_list'
+            API call with the provided device IP address, logs the response, and returns list containing ap device ips.
+        """
+
+        ap_device_list = []
+        for device_ip in device_ips:
+            try:
+                response = self.dnac._exec(
+                    family="devices",
+                    function='get_device_list',
+                    params={"managementIpAddress": device_ip}
+                )
+                response = response.get('response')
```

Comment:
can we simplify this, it will become more readable..
```
            response = response.get('response', [])

            if response and response[0].get('family', '') == "Unified AP":
                ap_device_list.append(device_ip)


```

Reply:
simplified and replace with above code.